test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -145,83 +154,119 @@ public void handle(final HttpExchange exchange) throws IOException {
                     .map(line -> line.substring(0, line.indexOf("</Latest>")))
                     .toList();
 
-                final ByteArrayOutputStream blob = new ByteArrayOutputStream();
-                for (String blockId : blockIds) {
-                    BytesReference block = blobs.remove(blockId);
-                    assert block != null;
-                    block.writeTo(blob);
-                }
-                blobs.put(exchange.getRequestURI().getPath(), new BytesArray(blob.toByteArray()));
+                mockAzureBlobStore.putBlockList(blobPath(exchange), blockIds, leaseId(exchange));
                 exchange.getResponseHeaders().add("x-ms-request-server-encrypted", "false");
                 exchange.sendResponseHeaders(RestStatus.CREATED.getStatus(), -1);
 
+            } else if (Regex.simpleMatch("PUT /" + account + "/" + container + "*comp=lease*", request)) {
+                // Lease Blob (https://learn.microsoft.com/en-us/rest/api/storageservices/lease-blob)
+                final String leaseAction = requireHeader(exchange, "x-ms-lease-action");
+
+                switch (leaseAction) {
+                    case "acquire" -> {
+                        final int leaseDurationSeconds = requireIntegerHeader(exchange, X_MS_LEASE_DURATION);
+                        final String proposedLeaseId = exchange.getRequestHeaders().getFirst(X_MS_PROPOSED_LEASE_ID);
+                        final String newLeaseId = mockAzureBlobStore.acquireLease(
+                            blobPath(exchange),
+                            leaseDurationSeconds,
+                            proposedLeaseId
+                        );
+                        exchange.getResponseHeaders().set(X_MS_LEASE_ID, newLeaseId);
+                        exchange.sendResponseHeaders(RestStatus.CREATED.getStatus(), -1);
+                    }
+                    case "release" -> {
+                        final String leaseId = requireHeader(exchange, X_MS_LEASE_ID);
+                        mockAzureBlobStore.releaseLease(blobPath(exchange), leaseId);
+                        exchange.sendResponseHeaders(RestStatus.OK.getStatus(), -1);
+                    }
+                    case "renew", "change", "break" -> {
+                        ExceptionsHelper.maybeDieOnAnotherThread(
+                            new AssertionError("Attempt was made to use not-implemented lease action: " + leaseAction)
+                        );
+                        throw new MockAzureBlobStore.AzureBlobStoreError(
+                            RestStatus.INTERNAL_SERVER_ERROR,
+                            "NotImplemented",
+                            "Attempted to use unsupported lease API: " + leaseAction
+                        );
+                    }
+                    default -> throw new MockAzureBlobStore.BadRequestException(
```

Comment:
I think we should fail the test here too

Reply:
Done in 0543928d82469192ddd7253d2fdf876b475fd619